pkg/liqo-controller-manager/offloadingStatus-controller/offloadingstatus_controller.go

```diff
@@ -0,0 +1,153 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package offloadingstatuscontroller
+
+import (
+	"context"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/klog"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+
+	offv1alpha1 "github.com/liqotech/liqo/apis/offloading/v1alpha1"
+	mapsv1alpha1 "github.com/liqotech/liqo/apis/virtualKubelet/v1alpha1"
+	liqoconst "github.com/liqotech/liqo/pkg/consts"
+	liqoutils "github.com/liqotech/liqo/pkg/utils"
+)
+
+// OffloadingStatusReconciler checks the status of all remote namespaces associated with this
+// namespaceOffloading Resource, and sets the global offloading status in according to the feedbacks received
+// from all remote clusters.
+type OffloadingStatusReconciler struct {
+	client.Client
+	Scheme      *runtime.Scheme
+	RequeueTime time.Duration
+}
+
+// Controller Ownership:
+// --> NamespaceOffloading.Status.RemoteConditions
+// --> NamespaceOffloading.Status.OffloadingPhase
+
+func setNamespaceOffloadingStatus(noff *offv1alpha1.NamespaceOffloading) {
```

Comment:
Would move this to a separate file, along with the other accessory functions

Reply:
ty